packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -389,6 +389,88 @@ it('should handle row clicks', async () => {
   expect(rows[2].classList).toContain('iui-selected');
   expect(onSelect).toHaveBeenCalledTimes(3);
   expect(onRowClick).toHaveBeenCalledTimes(3);
+
+  await user.keyboard('[/ControlLeft][ShiftLeft>]'); // Release Control and Press Shift (without releasing it)
+  await user.click(getByText(mockedData()[0].name)); // Perform a click with `shiftKey: true`
+
+  expect(rows[0].classList).toContain('iui-selected');
+  expect(rows[1].classList).toContain('iui-selected');
+  expect(rows[2].classList).toContain('iui-selected');
+  expect(onSelect).toHaveBeenCalledTimes(3 + (1 + 3)); // # times before + (reset + # selected rows from start to end)
+  expect(onRowClick).toHaveBeenCalledTimes(4);
+});
+
+it('should handle sub-rows shift click selection', async () => {
+  const onSelect = jest.fn();
+  const onRowClick = jest.fn();
+  const data = mockedSubRowsData();
+  const { container, getByText } = renderComponent({
+    data,
+    onSelect,
+    onRowClick,
+    isSelectable: true,
+  });
+
+  let rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(3);
+
+  const user = userEvent.setup();
+  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
+  await user.click(getByText(data[1].name)); // [shiftKey: true]
+
+  // By default, when no row is selected before shift click, start selecting from first row to clicked row
+  expect(rows[0].classList).toContain('iui-selected');
+  expect(rows[1].classList).toContain('iui-selected');
+  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(onSelect).toHaveBeenCalledTimes(1 + 2); // reset + # selected rows from start to end
+  expect(onRowClick).toHaveBeenCalledTimes(1);
+
+  await expandAll(container);
+
+  rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(10);
+
+  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
+  await user.click(getByText(data[0].subRows[1].subRows[0].name)); // [shiftKey: true]
+
+  let checkboxes = container.querySelectorAll<HTMLInputElement>(
+    '.iui-table-body .iui-checkbox',
+  );
+  let selectedIndices = [1, 3];
+  Array.from(checkboxes).forEach((checkbox, index) => {
+    expect(!!checkbox.checked).toBe(selectedIndices.includes(index));
+  });
+
+  expect(onSelect).toHaveBeenCalledTimes(3 + (1 + 2)); // # times before + (reset + # selected rows from start to end)
```

Comment:
This `toHaveBeenCalledTimes` calculations can lead to faulty tests. Maybe there is a better way to test function call. Eg. `toHaveBeenCalledLastWith`? It would test if specific row is selected and not just random row.

Reply:
**[DONE]** Thanks for idea. After implementing Vilius's idea for the shift select (branch `vilius/shift-select-demo`), `onSelect()` is only called once per total shift select. So not there are no more required calculations for the # of times `onSelect()` is called. The number of times `onSelect()` is called is just one more than the previous one.